### PR TITLE
[Bug] Fixes Transform/Imposter not updating type/battle stat changes immediately; set move PP to 5 when transforming

### PR DESCRIPTION
**Pull request by schmidtc1**
## What are the changes?

Transform/Imposter now properly update type and battle stat changes on use/switch in, respectively.
Transform/Imposter now properly set move PP to 5 (if the move's PP is greater than 5) when transforming

## Why am I doing these changes?

It's a bug.
#2564 
Transform/Imposter do not currently set the move PP to 5, as stated in [Bulbapedia](https://bulbapedia.bulbagarden.net/wiki/Transform_(move)#:~:text=When%20a%20move%20is%20copied%2C%20it%20will%20have%205%20PP%2C%20unless%20its%20maximum%20PP%20is%20less%20(such%20as%20Sketch).) 

## What did change?

Added `user.updateInfo()` to the end of the `TransformAttr` and the `PostSummonTransformAbAttr` to refresh the user's changes.
Placed the copy code of `PostSummonTransformAbAttr` into a `Promise` to mirror `TransformAttr`. 
Transform/Imposter now set the move's pp to 5 using a negative ppUp value, if the move's pp is greater than 5.

### Screenshots/Videos

Before (Transform)

https://github.com/user-attachments/assets/ff9e0561-e45c-4115-aaeb-5229523a66db

After (Transform)

https://github.com/user-attachments/assets/2473d6e0-a9a4-461a-aa16-0e6110f48f06

Before (Imposter)

https://github.com/user-attachments/assets/d6c58364-acec-45ee-bded-d12734086f50

After (Imposter)

https://github.com/user-attachments/assets/aeba731c-0d3a-4e56-9440-87591183154e

## How to test the changes?

Use src/overrides.ts to give your pokemon Transform or Imposter, check stat changes, type changes, and PP value are updated correctly.

## Checklist
- [x] **I'm using `beta` as my base branch**
- [x] There is no overlap with another PR?
- [x] The PR is self-contained and cannot be split into smaller PRs?
- [x] Have I provided a clear explanation of the changes?
- [ ] Have I considered writing automated tests for the issue?
- [x] Have I tested the changes (manually)?
    - [x] Are all unit tests still passing? (`npm run test`)
- [x] Are the changes visual?
  - [x] Have I provided screenshots/videos of the changes?

### CODE REVIEW

**Comment by schmidtc1** [comment]
> Could you resolve the merge conflicts?

Resolved, but I want to wait on #2699 before I continue further development, as that PR likely fixes transform/imposter's stat change bugs.

File: src/data/ability.ts [hunk 1]
```diff
@@ -2450,18 +2451,23 @@ export class PostSummonTransformAbAttr extends PostSummonAbAttr {
       pokemon.setStatStage(s, target.getStatStage(s));
     }
 
-    pokemon.summonData.moveset = target.getMoveset().map(m => new PokemonMove(m?.moveId ?? Moves.NONE, m?.ppUsed, m?.ppUp));
+    pokemon.summonData.moveset = target.getMoveset().map(m => {
+      const pp = m?.getMove().pp!;
```
Comment (DayKev) [inline src/data/ability.ts:2455]:
```suggestion
      const pp = m?.getMove().pp ?? 0;
```

File: src/data/move.ts [hunk 2]
```diff
@@ -6418,39 +6418,46 @@ export class SuppressAbilitiesIfActedAttr extends MoveEffectAttr {
...
-      }
+    // Copy all stat stages
+    for (const s of BATTLE_STATS) {
+      user.setStatStage(s, target.getStatStage(s));
+    }
 
-      user.summonData.moveset = target.getMoveset().map(m => new PokemonMove(m?.moveId!, m?.ppUsed, m?.ppUp)); // TODO: is this bang correct?
-      user.summonData.types = target.getTypes();
+    user.summonData.moveset = target.getMoveset().map(m => {
+      const pp = m?.getMove().pp!;
```
Comment (DayKev) [inline src/data/move.ts:6444]:
```suggestion
      const pp = m?.getMove().pp ?? 0;
```

**Review by DayKev — COMMENTED** [review]
Needs test(s), otherwise seems good.

**Comment by schmidtc1** [comment]
> Needs test(s), otherwise seems good.



> Needs test(s), otherwise seems good.

Added tests for the pp change, not sure if I need to add unit tests for the visual fix as well?

**Comment by DayKev** [comment]
You can't really make tests for visual changes so no need to worry about those.

**Review by DayKev — APPROVED** [review]

**Pull request merged**